src/App.tsx

```diff
@@ -9,36 +9,155 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { Comment } from './types/Comment';
+import { CurrentUser } from './types/CurrentUser';
+import { CurrentPost } from './types/CurrentPost';
+import { ApiTypes } from './utils/ApiTypes';
+import { ErrorType } from './utils/ErrorType';
+
+import {
+  getUsers, getPosts, getComments, post, removeComment,
+} from './api';
+
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[] | null>(null);
+  const [comments, setComments] = useState<Comment[] | null>(null);
+  const [error, setError] = useState<ErrorType>(ErrorType.INITIAL);
+  const [isSelectClicked, setIsSelectClicked] = useState<boolean>(false);
+  const [isCommentClicked, setIsCommentClicked] = useState<boolean>(false);
+  const [currentUser, setCurrentUser] = useState<CurrentUser | null>(null);
```

Comment:
```suggestion
  const [currentUserId, setCurrentUserId] = useState<number | null>(null);
```
Make the same with post